awswrangler/_distributed.py

```diff
@@ -57,16 +58,21 @@ def get(cls) -> Enum:
         str
             The distribution engine configured.
         """
-        return cls._enum if cls._enum else cls.get_installed()
+        return cls._engine if cls._engine else cls.get_installed()
 
     @classmethod
     def set(cls, name: str) -> None:
         """Set the distribution engine."""
-        cls._enum = EngineEnum._member_map_[name.upper()]  # pylint: disable=protected-access,no-member
+        cls._engine = EngineEnum._member_map_[  # pylint: disable=protected-access,no-member
+            name.upper()
+        ]  # type: ignore
 
     @classmethod
     def dispatch_func(cls, source_func: Callable[..., Any], value: Optional[Any] = None) -> Callable[..., Any]:
         """Dispatch a func based on value or the distribution engine and the source function."""
+        if not cls.is_initialized(cls.get().value):
```

Comment:
@jaidisido What do you think, should we keep this? I'm conflicted because if somebody switches and engine by just using `wr.engine.set`, this will ensure that the right engine is initialized. On the other hand, the root issue here is that we have three methods that can be used to switch engines:
1. `engine.set`
2. `engine.register`
3. `engine.initialize`

Do we want to change these so that only `initialize` is publicly accessible?

Reply:
Personally I would keep this simple and remove this conditional for now until we have clearly defined how we want to handle lazy initialisation? But happy to discuss